File: graphql-dgs-codegen-core/src/integTest/java/com/netflix/graphql/dgs/codegen/java/testcases/dataclasses/dataClassWithDeclaredScalars/expected/types/Entity.java
package com.netflix.graphql.dgs.codegen.java.testcases.dataclasses.dataClassWithDeclaredScalars.expected.types;

import java.lang.Long;
import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.time.OffsetDateTime;
import java.util.Objects;

public class Entity {
  private Long _long;

  private OffsetDateTime dateTime;

  public Entity() {
  }

  public Entity(Long _long, OffsetDateTime dateTime) {
    this._long = _long;
    this.dateTime = dateTime;
  }

  public Long getLong() {
    return _long;
  }

  public void setLong(Long _long) {
    this._long = _long;
  }

  public OffsetDateTime getDateTime() {
    return dateTime;
  }

  public void setDateTime(OffsetDateTime dateTime) {
    this.dateTime = dateTime;
  }

  @Override
  public String toString() {
    return "Entity{long='" + _long + "', dateTime='" + dateTime + "'}";
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    Entity that = (Entity) o;
    return Objects.equals(_long, that._long) &&
        Objects.equals(dateTime, that.dateTime);
  }

  @Override
  public int hashCode() {
    return Objects.hash(_long, dateTime);
  }

  public static Builder newBuilder() {
    return new Builder();
  }

  public static class Builder {
    private Long _long;

    private OffsetDateTime dateTime;

    public Entity build() {
      Entity result = new Entity();
      result._long = this._long;
      result.dateTime = this.dateTime;
      return result;
    }

    public Builder _long(Long _long) {
      this._long = _long;
      return this;
    }

    public Builder dateTime(OffsetDateTime dateTime) {
      this.dateTime = dateTime;
      return this;
    }
  }
}

File: graphql-dgs-codegen-core/src/integTest/kotlin/com/netflix/graphql/dgs/codegen/cases/dataClassWithMappedTypes/expected/types/Entity.kt
package com.netflix.graphql.dgs.codegen.cases.dataClassWithMappedTypes.expected.types

import com.fasterxml.jackson.`annotation`.JsonIgnoreProperties
import com.fasterxml.jackson.`annotation`.JsonProperty
import com.fasterxml.jackson.`annotation`.JsonTypeInfo
import com.fasterxml.jackson.databind.`annotation`.JsonDeserialize
import com.fasterxml.jackson.databind.`annotation`.JsonPOJOBuilder
import java.lang.IllegalStateException
import java.time.OffsetDateTime
import kotlin.Long
import kotlin.jvm.JvmName

@JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
@JsonDeserialize(builder = Entity.Builder::class)
public class Entity(
    long: () -> Long? = longDefault,
    dateTime: () -> OffsetDateTime? = dateTimeDefault
) {
    private val _long: () -> Long? = long

    private val _dateTime: () -> OffsetDateTime? = dateTime

    @get:JvmName("getLong")
    public val long: Long?
        get() = _long.invoke()

    @get:JvmName("getDateTime")
    public val dateTime: OffsetDateTime?
        get() = _dateTime.invoke()

    public companion object {
        private val longDefault: () -> Long? = { throw IllegalStateException("Field `long` was not requested") }

        private val dateTimeDefault: () -> OffsetDateTime? = { throw IllegalStateException("Field `dateTime` was not requested") }
    }

    @JsonPOJOBuilder
    @JsonIgnoreProperties("__typename")
    public class Builder {
        private var long: () -> Long? = longDefault

        private var dateTime: () -> OffsetDateTime? = dateTimeDefault

        @JsonProperty("long")
        public fun withLong(long: Long?): Builder = this.apply {
            this.long = { long }
        }

        @JsonProperty("dateTime")
        public fun withDateTime(dateTime: OffsetDateTime?): Builder = this.apply {
            this.dateTime = { dateTime }
        }

        public fun build() = Entity(
            long = long,
            dateTime = dateTime
        )
    }
}
